crawl-ref/source/mon-act.cc

```diff
@@ -2222,6 +2222,63 @@ static void _torpor_snail_slow(monster* mons)
     }
 }
 
+/**
+ *  Make this monster use the magic of a holy lantern. (Vaud).
+ *
+ *  @param mons       The monster holding the lantern.
+ */
+static void _vaud_lantern_burn(monster *mons)
+{
+    //hurt player
+    //check in range and in LOS
+    if (cell_see_cell(you.pos(), mons->pos(), LOS_SOLID_SEE)
+        && grid_distance(mons->pos(), you.pos()) < 4)
+    {
+        //check divine protection
+        if (!is_good_god(you.religion)
+            && !is_sanctuary(you.pos()))
+        {
+            //deal holy flavoured damage
+            targeter_radius hitfunc(mons, LOS_SOLID);
+            flash_view_delay(UA_MONSTER, YELLOW, 100, &hitfunc);
+
+            if (you.undead_state() != US_ALIVE)
+            {
+                mprf("Vaud's lantern begins to melt your undead flesh!");
```

Comment:
Is it fun to have a unique that's only threatening if the player is undead? It seems hard to balance so that this enemy is threatening if the player is alive without being insanely lethal against undead players, but running into a unique that's a pushover for the vast majority of characters doesn't sound great, either.

How core is the goal of this being "a scary unique to encounter should the player be undead"? Would it make more sense to refocus around the idea of the passive aoe lantern damage and retheme away from the holy aspect?